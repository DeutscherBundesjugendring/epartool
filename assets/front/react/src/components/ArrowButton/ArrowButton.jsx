import React from 'react';


const ArrowButton = (props) => {
  let classes = null;

  if (props.type === 'followup-timeline-count') {
    classes = 'followup-link followup-timeline-count followup-sprite followup-sprite-timeline-count';
  }

  if (props.direction === 'left') {
    // Bed≈ôich Schindler <bedrich@visionapps.cz>, 28. 4. 2017 16:00
    // TODO: Add css class followup-link-right
    classes += ' followup-link-left';
  }

  return (
    <span className={classes}>
      {props.label}
    </span>
  );
};

ArrowButton.propTypes = {
  label: React.PropTypes.string.isRequired,
  type: React.PropTypes.oneOf(['followup-timeline-count']).isRequired,
  direction: React.PropTypes.oneOf(['left', 'right']).isRequired,

};


export default ArrowButton;
