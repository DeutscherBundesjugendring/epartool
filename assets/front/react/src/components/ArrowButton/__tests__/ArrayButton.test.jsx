import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import ArrowButton from '../ArrowButton';


describe('rendering', () => {
  it('renders correctly right', () => {
    const tree = shallow(
      <ArrowButton type="followup-timeline-count" label="1" direction="right" />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly left', () => {
    const tree = shallow(
      <ArrowButton type="followup-timeline-count" label="1" direction="left" />
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
