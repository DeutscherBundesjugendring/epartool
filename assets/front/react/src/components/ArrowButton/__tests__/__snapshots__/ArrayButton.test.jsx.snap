// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly left 1`] = `
<span
  className="followup-link followup-timeline-count followup-sprite followup-sprite-timeline-count followup-link-left"
>
  1
</span>
`;

exports[`rendering renders correctly right 1`] = `
<span
  className="followup-link followup-timeline-count followup-sprite followup-sprite-timeline-count"
>
  1
</span>
`;
